### PR TITLE
Adding Rotation property for Spawning Event

### PR DESCRIPTION
**Pull request by louis1706**

### CODE REVIEW

File: Exiled.Events/EventArgs/Player/SpawningEventArgs.cs [hunk 1]
```diff
@@ -51,6 +55,14 @@ public SpawningEventArgs(Player player, Vector3 position, PlayerRoleBase oldRole
         /// </remarks>
         public Vector3 Position { get; set; }
 
+        /// <summary>
+        ///     Gets or sets the <see cref="Player"/>'s spawning rotation.
+        /// </summary>
+        /// <remarks>
+        ///     Rotation will apply only for <see cref="FpcRole"/>.
+        /// </remarks>
+        public float Rotation { get; set; }
```
Comment (Thundermaker300) [inline Exiled.Events/EventArgs/Player/SpawningEventArgs.cs:64]:
```suggestion
        public float HorizontalRotation { get; set; }
```
Rename to HorizontalRotation to match the SpawnLocation APIs.

**Review by Thundermaker300 — CHANGES_REQUESTED** [review]

File: Exiled.Events/EventArgs/Player/SpawningEventArgs.cs [hunk 2]
```diff
@@ -22,19 +22,23 @@ public class SpawningEventArgs : IPlayerEvent
...
         /// <param name="oldRole">
         ///     the spawned player's old <see cref="PlayerRoleBase">role</see>.
         /// </param>
-        public SpawningEventArgs(Player player, Vector3 position, PlayerRoleBase oldRole)
+        public SpawningEventArgs(ReferenceHub hub, Vector3 position, float rotation, PlayerRoleBase oldRole)
         {
-            Player = player;
+            Player = Player.Get(hub);
             Position = position;
+            Rotation = rotation;
```
Comment (Thundermaker300) [inline Exiled.Events/EventArgs/Player/SpawningEventArgs.cs:41]:
```suggestion
            HorizontalRotation = rotation;
```

**Review by Thundermaker300 — CHANGES_REQUESTED** [review]

**Review by joker-119 — CHANGES_REQUESTED** [review]
Why remove the player check for spawning? This ensures the event isn't triggered by the server or a null played

File: Exiled.Events/Patches/Events/Player/Spawning.cs [hunk 3]
```diff
@@ -43,26 +43,29 @@ private static MethodInfo TargetMethod()
 
         private static bool Prefix(ReferenceHub hub, PlayerRoleBase prevRole, PlayerRoleBase newRole)
         {
-            if (Player.TryGet(hub, out Player player) && newRole.ServerSpawnReason != RoleChangeReason.Destroyed)
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/Spawning.cs:46]: Like Joker said, is this check not necessary anymore?

**Review by iopietro — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Player/Spawning.cs (hunk 3 above)
Comment (louis1706) [reply Exiled.Events/Patches/Events/Player/Spawning.cs:46]: I thinks it's prevent NRE and base game refuse Destroyed if i remind correctly

**Review by iopietro — APPROVED** [review]

**Review by Thundermaker300 — APPROVED** [review]

**Pull request merged**